src/HigherOrderExpectation.php

```diff
@@ -33,16 +33,24 @@ final class HigherOrderExpectation
     /**
      * @var string
      */
-    private $property;
+    private $name;
 
     /**
      * Creates a new higher order expectation.
      */
-    public function __construct(Expectation $original, string $property)
+    public function __construct(Expectation $original, string $name, bool $asMethod = false, ...$arguments)
```

Comment:
Could the `$asMethod` and `$arguments` be passed in a separate method?

```php
return (new HigherOrderExpectation($this, $method))->asMethod(...$parameters);
```

Reply:
I thought exactly this but went with parameters to allow the expectation set-up to happen in the constructor? 

I'm not sure of a clean way to defer the setup of the expectation in the class without having everywhere that calls `$this->expectation` to instead call `$this->expectation()` and have that method as:
```php
private function expectation()
{
   return $this->expectation ??= // Set up the expectation here
}
```

Reply:
@lukeraymonddowning - I've added static constructors as one way at solving this:
```php
return HigherOrderExpectation::forProperty($this, $name)

return HigherOrderExpectation::forMethod($this, $name, ...$parameters)
```
I'm not sure if static constructors are in fitting with this repos code style but I thought it was at least presenting it.

Reply:
Could we instead assume `asMethod` if the constructor received more than 2 arguments?

Reply:
@lukeraymonddowning *chef's kiss*. Great call, thank you 🔥